in-progress/8757-fees/design.md

```diff
@@ -0,0 +1,192 @@
+
+```math
+\text{congestion factor} = \text{congestion factor multiplier} * \text{exp}\left(\frac{\text{total mana used} + \text{previous excess mana}}{\text{target mana per block}}\right)
+```
+
+The base fee in wei/mana is then:
+
+```math
+\text{base wei per mana} = \text{congestion factor} * \text{cost of block in wei/mana}
+```
+
+
+## The cost of a transaction in the fee asset
+
+The amount of mana a transaction consumes is:
+
+```math
+\text{mana}_{tx} = \text{overhead mana per tx} + \sum_{op \in tx} \text{mana per operation}(op)
+```
+
+Therefore, the cost of a transaction's proposal, data publication, and verification at time $i$ is, denominated in the fee asset:
+
+```math
+\text{fee asset}_{tx} = \text{mana}_{tx} * \text{base wei per mana} * \text{fee asset per wei}
+```
+
+## Distribution of the fee asset
+
+When a transaction is included in a block in the pending chain, the L2 balance of the fee asset for the `fee_payer` of the transaction is reduced by the transactions's fee asset cost.
+
+When an epoch is proven, the fees for the epoch are paid out to each proposer for the block(s) they proposed.
+
+A fixed percentage of each block's fee asset fees are paid to the prover of the epoch. The percentage is determined based on the quote submitted by the prover for the epoch.
+
+This quote is submitted by the prover in epoch `i+1` to prove epoch `i`, and is thus claimed/submitted to L1 by a proposer in epoch `i+1`.
+
+In order to prevent a prover bribing a proposer into accepting a quote with an extremely high fee, the maximum fee a prover can submit for an epoch proof quote is `maximum_epoch_proof_quote_fee`.
+
+## Clarifications and considerations
+
+### Why is `blobs_per_block` a constant?
+
+It makes computation easier. Otherwise block builder would need to know the number of blobs consumed by the transaction while it is building.
+
+### L1 Congestion
+
+If L1 gas spikes during an epoch, this will be reflected in the TST cost of a transaction through factoring in the higher `wei_per_l1_gas` and `wei_per_l1_blob_gas` into the TST cost.
+
+However, the base fee is computed at the time of a block being proposed, based on the current L1 base fees. These may change between the block being proposed and the epoch being verified.
+
+Thus, it is possible for L1 gas prices to spike to a point where it is not profitable for a prover to submit their proof to L1. There are two retorts:
+1. The prover should bake in a risk premium to compensate them for this risk; it is similar to the risk that the price of TST relative to any other asset shifts against them.
+2. The prover can wait until the end of the epoch waiting for a less congested window to submit their proof to L1.
+
+### Not charging for DA separately
+
+Current thinking is that proving costs will dominate the cost of a transaction, so metering DA is not necessary.
```

Comment:
We are metering it. We are just throwing it into the mix to just have one unit to account for now and not keeping two different costs ready.